ldk/javascript/src/promisify.ts

```diff
@@ -8,14 +8,15 @@ function handleCaughtError(reject: (reason?: any) => void, error: Error) {
   reject(error);
 }
 
-export function promisifyMappedWithParam<TParam, TInternalOut, TExternalOut>(
-  param: TParam,
+export function promisifyMappedBothWithParams<TParamIn, TParamOut, TInternalOut, TExternalOut>(
+  param: TParamIn,
+  paramMap: Mapper<TParamIn, TParamOut>,
```

Comment:
Will it be simpler to call mapper and then call promisifyMappedWithParam() instead of expanding this layer?

Reply:
I still vote that this is not a sustainable pattern..

Reply:
@RusGraf I'm not sure what you mean by calling mapper here; can you clarify?

Reply:
Yes, what I mean is the reason why we calling mapper as part of the prdmosify in other places is that we are mapping return parameters and we are trying to resolve with the mapped value. I'm wondering if it will be better if we do not grow generic promisify layer by adding extra complexity. Since we are trying to map input parameter could we just do mapping upfront before we call promisify?

Something like this:
`export function create(whisper: NewWhisper): Promise<Whisper> {
  const internalWhisper = convertToInternalWhisper(whisper);
  return promisifyMappedWithParam(
    whisper,
    convertToExternalWhisper,
    oliveHelps.whisper.create,
  );
}`

Reply:
I don't think this is adding any material complexity, though this does add a small abstraction. If I didn't feel we'd be likely to be converting more values I wouldn't go for this either. But I think we are likely to be converting more incoming values and here's why I think this logic makes sense here:

Putting all of our whispers through a standard handling layer allows us to standardize the LDK <-> OH interactions and makes the code IMO more maintainable in the long term. 

What I'd like to prevent is a scenario where a few years down the line we have 100 different capabilities that are doing some mapping conversions.  We identify a need to time the mapping process for whatever reason. With a common function like this we can easily instrument that logic. Without a common function like this, we'd have to instrument each of the functions individually.

Another potential scenario is that we decide at some point that we want all errors that rise during the mapping of incoming parameters to reject a promise instead of throwing an error (or vice versa). With a common process abstraction we can change the process easily.

Reply:
Yeah, I'm in support of the intent. The angle I was looking at is that promisify is not following SRP which is making it less maintainable. But perhaps it is just my view on it and I don't think we have any functional issues here so we could move on.